src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -147,15 +145,90 @@ int goto_instrument_parse_optionst::doit()
     get_goto_program();
     instrument_goto_program();
 
-    if(cmdline.isset("unwind"))
     {
+      bool unwind=cmdline.isset("unwind");
+      bool unwindset=cmdline.isset("unwindset");
+      bool unwindset_file=cmdline.isset("unwindset-file");
 
-      int k = stoi(cmdline.get_value("unwind"));
+      if(unwindset && unwindset_file)
+        throw "only one of --unwindset and --unwindset-file supported at a "
+              "time";
 
-      goto_unwind(goto_functions, k);
+      if(unwind || unwindset || unwindset_file)
+      {
+        int k=-1;
 
-      goto_functions.update();
-      goto_functions.compute_loop_numbers();
+        if(unwind)
+          k=stoi(cmdline.get_value("unwind"));
```

Comment:
I'm a bit surprised this works without std::, but it has done before. Maybe use safe_string2int so that we can later put in std::stoi across the codebase?